cmd/krel/cmd/push.go

```diff
@@ -235,10 +241,65 @@ func runPushBuild(opts *pushBuildOptions) error {
 		return errors.Errorf("unable to identify specified bucket for artifacts: %s", releaseBucket)
 	}
 
-	// Check if bucket exists.
-	if _, err = bucket.Attrs(context.Background()); err != nil {
+	// Check if bucket exists and user has permissions
+	perms, err := bucket.IAM().TestPermissions(context.Background(), []string{"storage.objects.create"})
+	if err != nil {
 		return errors.Wrap(err, "Unable to find release artifact bucket")
 	}
+	if len(perms) != 1 {
+		return errors.Errorf("GCP user must have at least storage.objects.create permissions on bucket %s", releaseBucket)
+	}
+
+	if err = util.RemoveAndReplaceDir(release.GCSStagePath); err != nil {
+		return errors.Wrap(err, "Unable remove and replace GCS staging directory.")
+	}
+
+	// Copy release tarballs
+	if err = util.CopyDirContents(release.ReleaseTarsPath, release.GCSStagePath); err != nil {
+		return errors.Wrap(err, "Unable to copy source directory into destination")
+	}
+
+	// Prepare helpful GCP scripts
+	gciPath := release.GetGCIPath()
+	if err = util.CopyFile(filepath.Join(release.GCEPath, "configure-vm.sh"), filepath.Join(release.GCSStagePath, "extra/gce"), true); err != nil {
```

Comment:
Maybe place all of the scripts in a slice and range over them instead?
Also, is this going to verify the file exists first?

Reply:
For some of them it does, the last argument indicates whether to skip if the file does not exist or error. So for the `configure-vm.sh` we pass in `true` because we just want it staged if it exists.